File: 04-04-2024/Sum of all substrings of a number.cs
//{ Driver Code Starts
//Initial Template for C#


using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode
{

    class GFG
    {
        static void Main(string[] args)
        {
            int testcases;// Taking testcase as input
            testcases = Convert.ToInt32(Console.ReadLine());
            while (testcases-- > 0)// Looping through all testcases
            {
                string name = Console.ReadLine();
                Solution obj = new Solution();
                long res = obj.sumSubstrings(name);
                Console.Write(res+"\n");
            }

        }
    }
}

// } Driver Code Ends




//User function Template for C#

class Solution
{
     private const long Mod = 1000000007;
    //Complete this function
    //Function to find sum of all possible substrings of the given string.
    public long sumSubstrings(string s)
    {
        //Your code Start
          long ans = 0;
        long temp = 0;
        int n = s.Length;
        for (int i = 0; i < n; i++) {
            temp = temp * 10 + (i + 1) * (s[i] - '0');
            temp %= Mod;
            ans += temp;
            ans %= Mod;
        }
        return ans;
        //Your Code End
    }
}

File: 04-04-2024/Sum of all substrings of a number.py
#User function Template for python3

class Solution:
    # Function to find the sum of all possible substrings of the given string.
    def sumSubstrings(self, s):
        #Your Code Start
        mod = 1000000007
        ans = 0
        temp = 0
        n = len(s)

        for i in range(n):
            # Convert character to integer and calculate the contribution
            # of the digit to the total sum.
            temp = (temp * 10 + (i + 1) * int(s[i])) % mod
            ans = (ans + temp) % mod

        return ans
      #Your Code End
# code here


#{ 
 # Driver Code Starts
#Initial Template for Python 3
import atexit
import io
import sys

import sys
sys.setrecursionlimit(10**6)

# Contributed by : Nagendra Jha

if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases):
        s = str(input())
        ob=Solution()
        print(ob.sumSubstrings(s))
# } Driver Code Ends
